test/gtest/uct/ib/test_ib.cc

```diff
@@ -306,6 +306,92 @@ UCS_TEST_P(test_uct_ib_roce, local_subnet_only, "IB_ROCE_LOCAL_SUBNET=y")
 
 UCT_INSTANTIATE_IB_TEST_CASE(test_uct_ib_roce);
 
+class test_uct_ib_is_connected : public test_uct_ib {
+};
+
+UCS_TEST_P(test_uct_ib_is_connected, connected)
+{
+    uct_ep_is_connected_params_t params;
+    uct_iface_addr_t *iface_addr;
+    uct_device_addr_t *dev_addr;
+    uct_ep_addr_t *ep_addr;
+    uct_iface_attr_t iface_attr;
+
+    ASSERT_UCS_OK(uct_iface_query(m_e2->iface(), &iface_attr));
+    dev_addr   = (uct_device_addr_t*)malloc(iface_attr.device_addr_len);
+    iface_addr = (uct_iface_addr_t*)malloc(iface_attr.iface_addr_len);
+
+    memset(&params, 0, sizeof(params));
+    ASSERT_UCS_OK(uct_iface_get_address(m_e2->iface(), iface_addr));
+    ASSERT_UCS_OK(uct_iface_get_device_address(m_e2->iface(), dev_addr));
+
+    params.iface_addr  = iface_addr;
+    params.device_addr = dev_addr;
+    params.field_mask  = UCT_EP_IS_CONNECTED_FIELD_DEVICE_ADDR |
+                         UCT_EP_IS_CONNECTED_FIELD_IFACE_ADDR;
+
+    if (iface_attr.cap.flags & UCT_IFACE_FLAG_CONNECT_TO_EP) {
+        ep_addr = (uct_ep_addr_t*)malloc(iface_attr.ep_addr_len);
+        ASSERT_UCS_OK(uct_ep_get_address(m_e2->ep(0), ep_addr));
+        params.ep_addr     = ep_addr;
+        params.field_mask |= UCT_EP_IS_CONNECTED_FIELD_EP_ADDR;
+    }
+
+    EXPECT_TRUE(uct_ep_is_connected(m_e1->ep(0), &params));
+
+    free(dev_addr);
+    free(iface_addr);
+
+    if (iface_attr.cap.flags & UCT_IFACE_FLAG_CONNECT_TO_EP) {
+        free(ep_addr);
+    }
+}
+
+UCS_TEST_P(test_uct_ib_is_connected, not_connected)
```

Comment:
this test seems like massive code duplication.
We could just `EXPECT_FALSE(uct_ep_is_connected(m_e2->ep(0), &params));` in the previous test. Because m_e2 is connected to m_e1 (and not to itself)